pxtarget.json

```diff
@@ -95,6 +96,90 @@
         "breakBlock": true,
         "continueBlock": true
     },
+    "compile": {
+        "isNative": true,
+        "useUF2": true,
+        "webUSB": true,
+        "hasHex": true,
+        "saveAsPNG": true,
+        "deployDrives": ".*",
+        "deployFileMarker": "INFO_UF2.TXT",
+        "driveName": "DRIVE",
+        "openocdScript": "source [find interface/cmsis-dap.cfg]; set CHIPNAME at91samd21g18; source [find target/at91samdXX.cfg]",
+        "flashChecksumAddr": 8372,
+        "flashCodeAlign": 256,
+        "upgrades": [
+            {
+                "type": "package",
+                "map": {
+                    "robotis-cm": "robotis-cm-300"
+                }
+            }
+        ],
+        "patches": {
+            "0.0.0 - 0.6.8": [
+                {
+                    "type": "package",
+                    "map": {
+                        "arduino-mkr": "arduino-mkr1000"
+                    }
+                }
+            ]
+        }
+    },
+    "compileService": {
+        "buildEngine": "codal",
+        "codalTarget": {
+            "name": "missing",
+            "url": "https://github.com/microsoft/pxt-does-not-exist",
+            "branch": "v0",
+            "type": "git"
+        },
+        "codalBinary": "CIRCUIT_PLAYGROUND",
+        "yottaConfig": {
+            "pxt": {
+                "board": {
+                    "id": "BOARD_ID_CPLAY"
+                }
+            }
+        },
+        "githubCorePackage": "lancaster-university/codal",
+        "gittag": "v0.6.3",
+        "serviceId": "codal2cp",
+        "dockerImage": "pext/yotta:latest"
+    },
+    "variants": {
+        "cm300": {
+            "serial": {
+                "useHF2": true
+            },
+            "compile": {
+                "openocdScript": "source [find interface/cmsis-dap.cfg]; source [find target/nrf52.cfg]",
+                "flashChecksumAddr": 0,
+                "useUF2": true,
+                "webUSB": false,
+                "uf2Family": "0xada52840"                
+            },
+            "compileService": {
+                "codalTarget": {
+                    "name": "codal-cm300",
+                    "url": "https://github.com/ROBOTIS-GIT/codal-cm300.git",
+                    "branch": "develop",
+                    "type": "git"
+                },
+                "codalBinary": "CM300",
+                "serviceId": "codal2cm300",
```

Comment:
This service is not recognized by our cloud yet.

Reply:
Oh, I see.

Is it time-consuming or complicated to register a new service? Do you recommend us to use the existing ID (like codal2nrf52840) for the time being?

Reply:
use the existing one for now until i get your service registered.

Reply:
I modified serviceId to codal2nrf52840 on 271bc7e, but now I get a "length" error. 
What is the cause?